src/main/java/htsjdk/variant/variantcontext/writer/BCF2Writer.java

```diff
@@ -122,6 +122,12 @@
     private VCFHeader lastVCFHeaderOfUnparsedGenotypes = null;
     private boolean canPassOnUnparsedGenotypeDataForLastVCFHeader = false;
 
+    // is the header written to the output stream?
+    private boolean isHeaderWritten;
+
+    // is at least one part of body written to the output stream?
+    private boolean isBodyWritten;
+
```

Comment:
What's about using an `int` variable for different error messages? If different to -1, there is an error, and that can be switch to different messages...